trie/validation.py

```diff
@@ -18,6 +22,11 @@ def validate_length(value, length):
         raise ValidationError("Value is of length {0}.  Must be {1}".format(len(value), length))
 
 
+def validate_is_nibbles(value):
```

Comment:
This is where we'd benefit from having:

```python
class Nibble(enum.IntEnum):
    N0 = 0
    N1 = 1
    ...
    NE = 14
    NF = 15
```

Then we can either have things already typed as a nibble and thus not require any validation or just do `Nibble(value)` and it will raise a `TypeError` if the value is out of range.  For all other purposes, I believe each of the values for an `IntEnum` just evaluates as if it was an integer.

Reply:
Ok, tried it. Mixed feelings. I'll leave it in, because I'm pretty neutral on it. The biggest downside I see is that when you validate, you will get a confusing error if you try to use a byte-string instead of a tuple of nibbles for the prefix. So I ended up adding an extra tuple type check.

Reply:
I did a quick scan in the code but wasn't able to identify where/what you were referring to.  Would it be possible to address this by simply catching and re-raising a less confusing exception?